includes/cart/functions.php

```diff
@@ -233,8 +235,11 @@ function edd_add_to_cart( $download_id, $options = array() ) {
 		);
 	}
 
-	foreach ( $items as $item ) {
-		$to_add = apply_filters( 'edd_add_to_cart_item', $item );
+	foreach ( $items as &$item ) {
+		$item = apply_filters( 'edd_add_to_cart_item', $item );
+		$to_add = $item;
+		unset( $item );
```

Comment:
Hey @zackkatz I think I might not spoken correctly. the `unset` has to be right after the trailing `}` of the foreach:
```
foreach ( $items as &$item ) {
    // Do the awesome filter
    // Do the rest of the stuff
}
unset( $item );
```

Apparently with the referenced method that variable exists not outside of the foreach, as it doesn't destroy it.